src/Akka.Persistence.Sql/Query/SqlReadJournal.cs

```diff
@@ -59,6 +59,11 @@ public class SqlReadJournal :
             _system = system;
 
             var connFact = new AkkaPersistenceDataConnectionFactory(_readJournalConfig);
+            
+            // Fix for https://github.com/akkadotnet/Akka.Persistence.Sql/issues/344
+            var writeJournal = Persistence.Instance.Apply(system).JournalFor(writePluginId);
+            // we want to block, we want to crash if the journal is not available
+            var started = writeJournal.Ask<Initialized>(IsInitialized.Instance, TimeSpan.FromSeconds(5)).Result;
```

Comment:
Block and wait until startup is complete - this should cover any cases of this issue where queries still get executed before the write journal fully finishes initializing.